### PR TITLE
test-spec: Review and update Matter CI-matter-test

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
Matter CI triggers included too many components from subsys/net and that caused triggering Matter test plan on unrelated PRs.

CI-matter-test paths has been updated. Removed unnecessary paths to subsys/net.

### CODE REVIEW

**Review by lemrey — APPROVED** [review]
Thank you

**Review by kkasperczyk-no — APPROVED** [review]

File: .github/test-spec.yml [hunk 1]
```diff
@@ -297,14 +297,19 @@
...
   - "subsys/bootloader/**/*"
   - "subsys/partition_manager/**/*"
   - "subsys/nfc/**/*"
   - "subsys/nrf_rpc/**/*"
-  - "subsys/net/**/*"
+  - "subsys/dfu/**/*"
+  - "subsys/fw_info/*"
+  - "subsys/pcd/**/*"
+  - "subsys/bluetooth/controller/*"
+  - "subsys/net/openthread/*"
```
Comment (markaj-nordic) [inline .github/test-spec.yml:312]: Looks like we miss `subsys/net/l2/wifi` - this is related to Wi-Fi connection management and Wi-Fi shell support.

Comment (ArekBalysNordic) [reply .github/test-spec.yml:312]: I think you're looking at Zephyr directory. In this file there are only nrf-sdk related triggers, and I can't see subsys/net/l2 there.

Comment (markaj-nordic) [reply .github/test-spec.yml:312]: Sorry, I was looking at the 2.1.1 revision of NCS, indeed the l2/wifi has been removed recently:
https://github.com/nrfconnect/sdk-nrf/commit/2d700b9a1b4e812de0c4210f9c77b1647d326947

**Review by markaj-nordic — APPROVED** [review]

**Pull request merged**